src/encode.rs

```diff
@@ -10,7 +11,11 @@ use util::{zig_i32, zig_i64};
 /// be valid with regards to the schema. Schema are needed only to guide the
 /// encoding for complex type values.
 pub fn encode(value: &Value, schema: &Schema, buffer: &mut Vec<u8>) {
-    encode_ref(&value, schema, buffer)
+    encode_ref_inner(&value, &Arc::new(schema.clone()), buffer, &mut SchemaParseContext::new())
```

Comment:
Wouldn't cloning a schema every time a value is encoding end up being expensive?

We would have to run the benchmarks to see if there is a noticeable difference, but it seems like extra work is being done in this function